libcudacxx/test/libcudacxx/std/numerics/numeric.ops/inclusive.scan/inclusive_scan_op.pass.cpp

```diff
@@ -100,13 +100,15 @@ __host__ __device__ TEST_CONSTEXPR_CXX14 void basic_tests()
     }
   }
 
-  TEST_NV_DIAG_SUPPRESS(174) // expression has no effect
+#if !defined(TEST_COMPILER_NVHPC) // NVHPC seems unable to silence the warning
```

Comment:
@dkolsen-pgi I tried all versions of the #pragma suppression and would always get a warning here

Reply:
In my quick tests, `_Pragma("diag_suppress expr_has_no_effect")` suppresses the warning from both NVCC and NVC++ (including when NVC++ is the host compiler for NVCC).  But `_Pragma("nv_diag_suppress expr_has_no_effect")` only suppresses the warning from NVCC.  I don't know whether `TEST_NV_DIAG_SUPPRESS` expands to use `nv_diag_suppress` or `diag_suppress` when NVC++ is the host compiler for NVCC, but from the comment I am guessing it is `nv_diag_suppress`.

NVCC recognizes both `diag_suppress` and `nv_diag_suppress`.  NVC++ recognizes only `diag_suppress`.  The numeric arguments, such as `174`, are different between the two compilers (for historical reasons), but the named arguments should work in both.

Reply:
I tested it and it seems to not work if I put it in a macro, I believe this is the same issue as with our system header pragmas

Reply:
In my testing, putting the pragma in a macro works fine.
```cpp
$ cat noeffect.cpp
#define _TEST_TOSTRING2(x) #x
#define _TEST_TOSTRING(x)  _TEST_TOSTRING2(x)
#define TEST_NV_DIAG_SUPPRESS(WARNING) _Pragma(_TEST_TOSTRING(diag_suppress WARNING))
int main() {
  int a = 10, b = 42;
  TEST_NV_DIAG_SUPPRESS(expr_has_no_effect)
  {
    a + b;
  }
  return a + b;
}
$ nvc++ noeffect.cpp
$ nvcc -ccbin nvc++ -x cu noeffect.cpp
$ 
```

In the CI test environment, does `TEST_NV_DIAG_SUPPRESS` expand to use `diag_suppress` or `nv_diag_suppress` ?